### DISCUSSION TITLE
Regarding application of TP:Z:circular flag

### DISCUSSION THREAD

**Discussion by StamatisUCR** [ablab/spades#1357] [Q&A]
I'm looking at a metaspades assembly and have found a path in the graph that is labelled as circular. The path contains only one segment with a link to itself. It has a direct terminal repeat. However, the 55 base pairs that make up the DTR are also found in another segment, so the graph has a branch where the "circular" segment has a link both to itself and also this other segment. 

Why does spades tag the path as circular? Does it only consider the self-link, without considering the existence of branching paths due to the DTR existing in another segment? It seems as if the circularity is ambiguous, and after searching for the sequence of the segment of the DTR using blastx, the closest protein in an isolate is in a chromosome.

**Comment by asl** [comment]
Will you please post the graph itself? `TP:Z:circular` is used to mark exactly circular paths, so no branching / multiple links are possible.

**Reply by schorlton-bugseq** [reply]
Not the original author, but here's an example from the assembly graph of SRR11038968 run with SPAdes version: 4.0.0 after some gentle preprocessing: 
[assembly_graph_with_scaffolds.gfa.gz](https://github.com/user-attachments/files/17351881/assembly_graph_with_scaffolds.gfa.gz)

Here's an example of a circular path with additional links:
![image](https://github.com/user-attachments/assets/96773b76-c67c-4a2a-a777-b171b3724bcf)

Thanks for your help!
